components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/actuate/endpoint/CamelRoutesEndpoint.java

```diff
@@ -76,6 +74,11 @@ public void startRoute(String id) throws Exception {
         camelContext.getRouteController().startRoute(id);
     }
 
+    public void resetRoute(String id) throws Exception {
+        ManagedRouteMBean resetRoute = camelContext.getManagedRoute(id, ManagedRouteMBean.class);
+        resetRoute.reset();
```

Comment:
Not sure what other actuators do in such a case (are there dynamic actuators out there?), but I'm weary of 404 as it might be interpreted as a failure and for instance when run in a cloud environment might cause a restart. So I would go with 204.